### PR TITLE
Removes 2D MPFA-O code and tests

### LINKED ISSUE #46: Remove 2D MPFA-O code
It doesn't seem worthwhile to keep a separate 2D code when a 2D problem can be formulated as a 3D problem with  number of grid cells in one of the dimensions is one.

cc: @ghammond86

### PR DESCRIPTION
**Pull request by jeff-cohere**
This PR gets rid of all 2D logic in the MPFA-O code and merges the `src/mpfao/3D` source directory into `src/mpfao`. Related functions with `3D` in their names are renamed to simpler names. All 2D MPFA-O tests have been removed. This is the first step of the pruning process outlined in #200.

Closes #46

### CODE REVIEW

File: demo/mpfao/mpfao.cfg [hunk 1]
```diff
@@ -1,41 +1,18 @@
...
-	mpfao-prob4
 	mpfao-3d-prob
 #	mpfao-64xy-3z-wedge
 
-standard_parallel=
-	mpfao-prob3-np3
-	mpfao-prob3-perturb-np3
+#standard_parallel=
+#	mpfao-prob3-np3
+#	mpfao-prob3-perturb-np3
```
Comment (bishtgautam) [inline demo/mpfao/mpfao.cfg:8]: Delete these lines, as the tests have been deleted

**Comment by jeff-cohere** [comment]
Looks like we're losing some code coverage here. I guess that's not too surprising, considering the tests we've jettisoned. @bishtgautam , let me know if you're concerned about this in the short term. Further pruning might improve coverage.

File: demo/steady/steady.cfg [hunk 2]
```diff
@@ -16,7 +16,7 @@ standard=
 	steady-bdm-prob3
 	steady-bdm-prob4
 
-standard_parallel=steady-wy-np3
+#standard_parallel=steady-wy-np3
```
Comment (bishtgautam) [inline demo/steady/steady.cfg:19]: same comment as the previous one.

**Review by bishtgautam — APPROVED** [review]
I have two minor comments

**Comment by bishtgautam** [comment]
I'm not worried about the decrease in code coverage. At some point, we will have to add tests to improve code coverage.

**Pull request merged**